File: data/src/main/java/com/roman/kubik/songer/data/database/AppDatabase.java
package com.roman.kubik.songer.data.database;

import androidx.room.Database;
import androidx.room.RoomDatabase;

import com.roman.kubik.songer.data.category.CategoryDao;
import com.roman.kubik.songer.data.category.CategoryEntity;
import com.roman.kubik.songer.data.favourite.FavouriteDao;
import com.roman.kubik.songer.data.favourite.FavouriteEntity;
import com.roman.kubik.songer.data.history.HistoryDao;
import com.roman.kubik.songer.data.history.HistoryEntity;
import com.roman.kubik.songer.data.song.local.SongDao;
import com.roman.kubik.songer.data.song.local.SongEntity;

/**
 * Room database representation
 * Created by kubik on 1/14/18.
 */
@Database(entities = {SongEntity.class, CategoryEntity.class, FavouriteEntity.class, HistoryEntity.class}, version = 5, exportSchema = false)
public abstract class AppDatabase extends RoomDatabase {

    public abstract SongDao songDao();

    public abstract CategoryDao categoryDao();

    public abstract FavouriteDao favouriteDao();

    public abstract HistoryDao historyDao();

}

File: room/src/main/java/com/roman/kubik/songer/room/database/AppDatabase.kt
package com.roman.kubik.songer.room.database

import androidx.room.Database
import androidx.room.RoomDatabase
import com.roman.kubik.songer.room.category.CategoryEntity
import com.roman.kubik.songer.room.favourite.FavouriteEntity
import com.roman.kubik.songer.room.history.HistoryEntity
import com.roman.kubik.songer.room.song.SongDao
import com.roman.kubik.songer.room.song.SongEntity

/**
 * Room database representation
 */
@Database(entities = [SongEntity::class, CategoryEntity::class, FavouriteEntity::class, HistoryEntity::class], version = 5, exportSchema = false)
abstract class AppDatabase : RoomDatabase() {
    abstract fun songDao(): SongDao
//    abstract fun categoryDao(): CategoryDao?
//    abstract fun favouriteDao(): FavouriteDao?
//    abstract fun historyDao(): HistoryDao?
}